pytorch_lightning/trainer/properties.py

```diff
@@ -271,20 +336,87 @@ def checkpoint_callbacks(self) -> List[ModelCheckpoint]:
     def save_checkpoint(self, filepath, weights_only: bool = False):
         self.checkpoint_connector.save_checkpoint(filepath, weights_only)
 
+    @property
+    def model(self) -> Any:
+        """
+        The LightningModule, but possibly wrapped into DataParallel or DistributedDataParallel.
+        To access the pure LightningModule, use
+        :meth:`~pytorch_lightning.trainer.trainer.Trainer.lightning_module` instead.
+        """
+        return self.accelerator.model
+
+    @model.setter
+    def model(self, model: LightningModule):
```

Comment:
```suggestion
    def model(self, model: nn.Module):
```
LightningModule is too specific